configuration.md

```diff
@@ -43,55 +48,91 @@ If both are set then `etcd_endpoints` is used.
+As a convenience, the API location location can also be configured directly, e.g.
+```json
+{
+    "name": "any_name",
+    "type": "calico",
+    "kubernetes": {
+        "k8s_api_root": "http://127.0.0.1:8080"
+    }
+    "ipam": {
+        "type": "calico-ipam"
+    }
+}
+```
 
-To specify a policy, add the following block to the CNI network config:
+### Enabling Kubernetes Policy
+If you wish to use the Kubernetes NetworkPolicy API then you must set a policy type in the network config.
+There is a single supported policy type, `k8s` which uses the Kubernetes NetworkPolicy API in conjunction with the `calico/kube-policy-controller`.
 
 ```json
-"policy": {
-  "type": "<type>"
+{
+    "name": "any_name",
+    "type": "calico",
+    "policy": {
+      "type": "k8s",
+      "k8s_api_root": "http://127.0.0.1:8080"
+    }
+    "ipam": {
+        "type": "calico-ipam"
+    }
 }
 ```
 
-### Kubernetes API access details
-When using either policy type, the CNI plugin needs to be told how to access the Kubernetes API server in the `policy` section of the network config.
+Previous versions of the plugin (`v1.3.1` and earlier) supported an alternative type called [`k8s-annotations`](https://github.com/projectcalico/calico-containers/blob/v0.20.0/docs/cni/kubernetes/AnnotationPolicy.md) This uses annotations on pods to specify network policy but is no longer supported.
+
+### Deprecated ways of specifying Kubernetes API access details
+From the examples above, you can see that the `k8s_api_root` can appear in either the `kubernetes` or `policy` configuration blocks.
 * `k8s_api_root` (default `http://127.0.0.1:8080`)
 
-The CNI plugin may need to authenticate with the Kubernetes API server. The following methods are supported in the `policy` section of the CNI network config, none of which have default values.
-* `k8s_username`
-* `k8s_password`
+In addition, the following methods are supported in the `policy` section of the CNI network config only. None of them have default values.
 * `k8s_auth_token`
 * `k8s_client_certificate`
 * `k8s_client_key`
 * `k8s_certificate_authority`
-	* Verifying the API certificate against a CA only works if connecting to the API server using a hostname.
 
-The following methods are supported in the `kubernetes` section of the CNI network config.
-* `kubeconfig`
-	* Path to a Kubernetes `kubeconfig` file.
+### IPAM
+When using the CNI `host-local` IPAM plugin, a special value `usePodCidr` is allowed for the subnet field.  This tells the plugin to determine the subnet to use from the Kubernetes API based on the Node.podCIDR field.
+
 * `node_name`
     * The node name to use when looking up the `usePodCidr` value (defaults to current hostname)
 
 ```json
-"kubernetes": {
-    "kubeconfig": "/path/to/kubeconfig"
+{
+    "name": "any_name",
+    "type": "calico",
+    "kubernetes": {
```

Comment:
We discussed and I raised https://github.com/projectcalico/calico-cni/issues/161 which caputures (very briefly) the outcome of our conversation